### PR TITLE
Test object lock no overwrite in Compliance and Governance mode

### PR DESCRIPTION
**Pull request by TejasC88**
This PR addresses  BZ 1890843
CEPH-83574059 - Negative - Verify the objects cannot be overwritten.

Pass log : http://magna002.ceph.redhat.com/ceph-qe-logs/tejas/Object_lock_no_overwrite.txt

### CODE REVIEW

**Review by viduship — COMMENTED** [review]
Please address the tox failures

**Comment by TejasC88** [comment]
> Please address the tox failures

Sorry about this , request you to review now.

File: rgw/v2/tests/s3_swift/test_object_lock_no_overwrite.py [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+                        )
+                        raise AccessDeniedObjectDeleted("Access denied object deleted")
+                    except boto3exception.ClientError as e:
+                        expected_code = "AccessDenied"
+                        actual_code = e.response["Error"]["Code"]
+                        assert (
+                            actual_code == expected_code
+                        ), "Expected: {expected_code}, Actual: {actual_code}"
+                log.info("Unable to delete the object in retention period")
+                log.info("Delete key2 with bypassgovernanceretention")
```
[the remark this answers is not in the document]
Comment (TejasC88) [reply rgw/v2/tests/s3_swift/test_object_lock_no_overwrite.py:177]: @mkasturi18 we are trying out both scenarios in this script itself as part of the BZ, the above try except block is for BypassGovernanceRetention=False , and in this step we are mentioning it as true

[the remark this answers is not in the document]
Comment (mkasturi18) [reply rgw/v2/tests/s3_swift/test_object_lock_no_overwrite.py:177]: ack @TejasC88

**Review by mkasturi18 — COMMENTED** [review]
PR looks good

File: rgw/v2/tests/s3_swift/configs/test_object_lock_no_overwrite.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# upload type: non multipart
+# Related BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1890843
+# script: test_object_lock_no_overwrite.py - Test object lock configuration for bucket
```
Comment (ckulal) [inline rgw/v2/tests/s3_swift/configs/test_object_lock_no_overwrite.yaml:3]: Please include polarian Id aswell

Comment (TejasC88) [reply rgw/v2/tests/s3_swift/configs/test_object_lock_no_overwrite.yaml:3]: @ckulal I have added code for BZ 2252336 as well , request you to review

**Comment by TejasC88** [comment]
> PR looks good

I have added code for BZ 2252336 , request you to review that also.

**Review by anrao19 — DISMISSED** [review]

**Review by hmaheswa — DISMISSED** [review]

**Review by mkasturi18 — APPROVED** [review]
Changes looks good.

**Pull request merged**